pgjdbc/src/main/java/org/postgresql/jdbc/PgCallableStatement.java

```diff
@@ -691,7 +696,132 @@ public Reader getCharacterStream(String parameterName) throws SQLException {
   }
 
   public <T> T getObject(int parameterIndex, Class<T> type) throws SQLException {
-    if (type == ResultSet.class) {
+    if (type == null) {
+      throw new SQLException("type is null");
+    }
+    checkIndex(parameterIndex);
+    int sqlType = functionReturnType[parameterIndex - 1];
+    if (type == BigDecimal.class) {
+      if (sqlType == Types.NUMERIC || sqlType == Types.DECIMAL) {
+        return type.cast(getBigDecimal(parameterIndex));
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == String.class) {
+      if (sqlType == Types.CHAR || sqlType == Types.VARCHAR) {
+        return type.cast(getString(parameterIndex));
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == Boolean.class) {
+      if (sqlType == Types.BOOLEAN || sqlType == Types.BIT) {
+        boolean booleanValue = getBoolean(parameterIndex);
+        if (wasNull()) {
+          return null;
+        }
+        return type.cast(booleanValue);
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == Integer.class) {
+      if (sqlType == Types.SMALLINT || sqlType == Types.INTEGER) {
```

Comment:
AFAIK Postgres has no `TINYINT` so I don't think it's missing.

If you check [getByte](https://github.com/pgjdbc/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/jdbc/PgCallableStatement.java#L267) and [getShort](https://github.com/pgjdbc/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/jdbc/PgCallableStatement.java#L279) `SMALLINT`s are backed by an `Integer` so offering the conversion here is fine IMO. Keep in mind the caller asked for an `Integer` in this case so returning a `Short` would be a no-go, the only other option would be throwing and exception.